IliasDownloaderUniMA/__init__.py

```diff
@@ -127,6 +173,44 @@ def addCourses(self, *iliasids):
 			self.addCourse(iliasid)
 
 
+	def addAllUserCourses(self, excluded_ids=[]):
+		"""
+		Extracts the users subscribed courses and adds them to the course list.
+
+		:param      excluded_ids:  optional ilias ids to ignore
+		:type       excluded_ids:  list
+
+		:returns:   number of courses added
+		:rtype:     int
+		"""
+
+		num_added = 0
+		courses = self.extractUserCourses()
+
+		for course in courses:
+			iliasid = course["iliasid"]
+
+			if iliasid not in excluded_ids:
+				self.addCourse(iliasid)
+				num_added += 1
+
+		return num_added
+
+
+	def cleanCourseName(self, course_name):
+		"""
+		Cleans up a course name to remove unwanted data.
+
+		:param      course_name:  raw course_name(from login_soup, or request)
+		:type       course_name:  str
+
+		:returns:   clean course name
+		:rtype:     str
+		"""
+
+		return re.sub(r"\[.*\] ", "", course_name).strip()
```

Comment:
One-liner and only used once. Should be removed. Using it with a comment should be sufficient.

Reply:
It's actually used in two places,  addAllCourses(or whatever it's name will be in the end) and addCourse.
Didn't want to paste the RegEx in two places, so I went for the helper method.
I also did these because of the design philosophy in mind, look at my newest answer for that one.